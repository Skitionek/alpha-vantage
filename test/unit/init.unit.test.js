
import AlphaVantage from "../../src";

const env = process.env;
// Clear the current environment variables for testing.
process.env = {};

it(`initialization without a config throws an error`, () => {
	expect(()=>new AlphaVantage()).toThrowErrorMatchingSnapshot();
});

it(`initialization without an api key throws an error`, () => {
	expect(()=>new AlphaVantage()).toThrowErrorMatchingSnapshot();
});

it(`initialization without an api key, but with env key works`, () => {
	expect.assertions(1);
	try {
		process.env = Object.assign(env,{AV_KEY:'demo'});
		const alpha = new AlphaVantage();
		expect(alpha).toBeDefined();
	} catch (e) {
		expect(e).toBeUndefined();
	}
});
