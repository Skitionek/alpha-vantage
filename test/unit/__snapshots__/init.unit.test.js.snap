// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialization without a config throws an error 1`] = `"Missing Alpha Vantage config settings: key"`;

exports[`initialization without an api key throws an error 1`] = `"Missing Alpha Vantage config settings: key"`;
